### DISCUSSION TITLE
Generating Tweet Links from Content

### DISCUSSION THREAD

**Discussion by scco** [dcgtc/dgrants#315] [Architecture & Design]
this needs maybe a little bit of a discussion before we can make it a good issue. it also turns out it might be the best to add a share feature after all URLs, Routing is "final".

There is a pretty straight forward way of building a custom Tweet from a Link : 

`<a href="https://twitter.com/intent/tweet?text=description_variable&url=url_variable">Tweet</a>`

So that means, we do something like this : 

```
<!-- Tweet -->
<a target="_blank" :href="'https://twitter.com/intent/tweet?text=' + description_variable + '&url=https://grants.gtcdao.net' + $route.path "  class="flex items-center gap-x-2 cursor-pointer group ml-auto">
 <TweetIcon class="icon icon-primary icon-small" />
 <span class="text-grey-400 group-hover:text-grey-500">Tweet</span>
</a>
```

I would need help to build that :href on various pages.

### tweet a users  cart
the purpose of the tweet function in the cart is, to tweet a cart so other users can see a users cart and probably decide to takeover his items.  we do not have any kind of this features ready - e.g. build urls for this like
/cart/ids:34,125,65,347,45,157,451  
and also no interface ready for "add items to your own cart from an external url link ... "
we do not want other users to see their cart gets populated with another users selected items on click! 
it seems this import/export feature needs a bit more time and need to get kicked out from MVP.
 

### tweet a round
i am not sure about the url . right now it seems to be like 
`http://localhost:3000/#/dgrants/rounds/0x8B4091997E3EbB87be90cEd3e50d8Bb27e1DC742`
and our staging is  like this 
`https://staging.grants.gtcdao.net/`
so i assume a direct link to a round would be 
`https://grants.gtcdao.net/#/dgrants/rounds/0x8B4091997E3EbB87be90cEd3e50d8Bb27e1DC742`
for some reason when i tryed to add this string as variable to the tweet , it just cut on the /#/ in twitter ... 

### tweet a grant
should be the same logic - that we would like to share the Grant Name + the Direct URL to the Grant

**Comment by phutchins** [comment]
I've moved this to a discussion until we land on a solution that we can create a clear issue out of.

In the first iteration of the share via tweet feature, we should keep it fairly simple and avoid additional requirements where possible.

**UI Tweet/Share Implementation**
- After checking out, once on the "thank you for your contribution" page, a user should be shown a button to Share/Tweet about their contribution(s)
  - Content: I just donated to public goods on Gitcoin! Here are some of the grants that I supported... [list grants depending on how many we can fit]
  - bonus if we can add a link to take someone new to dGrants and pre-populate their cart with the same grants
- Each grant page should have a button to Share/Tweet that specific grant out to Twitter
  - Content: Support public goods by donating to this Grant on Gitcoin! [show title of grant as a link]
